app/models/attachment.rb

```diff
@@ -175,7 +175,9 @@ def set_file_size(file)
   end
 
   def set_content_type(file)
-    self.content_type = self.class.content_type_for(file.path) if content_type.blank?
+    return if content_type.present?
+
+    self.content_type = file.try(:content_type) || self.class.content_type_for(file.path)
```

Comment:
Some file interfaces (e.g. `UploadedFile`) have a content_type already set for us so we don't need to retrieve it manually. The mail attachments for example makes use of that. This became apparent after using the CreateService from the mail_handler.